src/main/java/org/ehcache/sizeof/ObjectGraphWalker.java

```diff
@@ -263,6 +263,14 @@ private static void nullSafeAdd(final Deque<Object> toVisit, final Object o) {
                         LOG.error("Security settings prevent Ehcache from accessing the subgraph beneath '{}'" +
                                   " - cache sizes may be underestimated as a result", field, e);
                         continue;
+                    } catch (RuntimeException e) {
+                        // new class (jdk9+ only) can be thrown
+                        if ("InaccessibleObjectException".equals(e.getClass().getSimpleName())) {
```

Comment:
I'd rather see us use the FQCN to do this comparison.

Reply:
```suggestion
                        if ("java.lang.reflect.InaccessibleObjectException".equals(e.getClass().getCanonicalName())) {
```

Reply:
A simple `getName()` should suffice here and would result in a simple string comparison. `getCanonicalName()` results in an uncached sequence of conditional checks which ultimately calls `getName()` in the only case we care about.

Reply:
```suggestion
                        if ("java.lang.reflect.InaccessibleObjectException".equals(e.getClass().getName())) {
```

Reply:
Would still like to see `getCanonicalName()` replaced with plain `getName()`

Reply:
?

Reply:
```suggestion
                        if ("java.lang.reflect.InaccessibleObjectException".equals(e.getClass().getName())) {
```

Reply:
```suggestion
                        if ("java.lang.reflect.InaccessibleObjectException".equals(e.getClass().getName())) {
--



```